### ISSUE TITLE
same output for every input

### ISSUE THREAD

**Issue by dharma-kc** [keithito/tacotron#36]
I was able to train the system and the loss got converged to 0.09. But when I test with every input, it produces the same output. Could you please guide me to the right direction. Thank you so much.

**Comment by keithito** [comment]
Can you share more about your training data and hyperparameters?

**Comment by dharma-kc** [comment]
I think its because of the utf-8 and unicode. I have my texts written in utf-8 format. Following is the one sample. 

1|﻿म कमल हो|म कमल हो

And I have made following changes.

non_Eng_charset = '﻿﻿अआइईउऊएऐओऔअंअ:ककाकिकीकुकूकेकैकोकौकंक:खखाखिखीखुखूखेखैखोखौखंख:गगागिगीगुगूगेगैगोगौगंग:घघाघिघीघुघूघेघैघोघौघंघ:ङङाङिङीङुङूङेङैङोङौङंङ:चचाचिची चुचूचेचैचोचौचंच:छछाछिछीछुछूछेछैछोछौछंछ:जजाजिजीजुजूजेजैजोजौजंज:झझाझिझीझुझूझेझैझोझौझंझ:ञञाञिञीञुञूञेञैञोञौञंञ:टटाटिटीटुटूटेटैटोटौटंट:ठठाठिठीठुठूठेठैठोठौठंठ:डडाडिडीडुडूडेडैडोडौडं ड:ढढाढिढीढुढूढेढैढोढौढंढ:णणाणिणीणुणूणेणैणोणौणंण:ततातितीतुतूतेतैतोतौतंत:थथाथिथीथुथूथेथैथोथौथंथ:ददादिदीदुदूदेदैदोदौदंद:धधाधिधीधुधूधेधैधोधौधंध:ननानिनीनुनूनेनैनोनौनंन:पपापिपीपुपूपेपैपोपौपंप:फ फाफिफीफुफूफेफैफोफौफंफ:बबाबिबीबुबूबेबैबोबौबंब:भभाभिभीभुभूभेभैभोभौभंभ:ममामिमीमुमूमेमैमोमौमंम:ययायियीयुयूयेयैयोयौयंय:ररारिरीरुरूरेरैरोरौरंर:ललालिलीलुलूलेलैलोलौलंल:ववाविवीवुवूवेवैवोवौवंव:ससासिसीसुसू सेसैसोसौसंस:शशाशिशीशुशूशेशैशोशौशंश:षषाषिषीषषूषेषैषोषौषंष:हहाहिहीहुहूहेहैहोहौहंह:क्षक्षक्षिक्षीक्षुक्षूक्षेक्षैक्षोक्षौक्षंक्ष:त्रत्रात्रित्रीत्रुत्रूत्रेत्रैत्रोत्रौत्रंत्र:ज्ञज्ञाज्ञिज्ञीज्ञुज्ञूज्ञेज्ञैज्ञोज्ञौज्ञंज्ञ:'

_valid_input_chars = _uppercase + _lowercase + _punctuation + _space + non_Eng_charset
in textinput.py. When I use the following line:

text = unicodedata.normalize('NFKD', text).encode('ascii', 'ignore').decode()

It trains and produces same output for every input. But when I replace this line with following line:

text = unicodedata.normalize('NFKD', text).encode('utf-8', 'ignore').decode()

It gives me following error:

![screenshot from 2017-09-03 13-15-10](https://user-images.githubusercontent.com/14134622/30001010-736d00e6-90aa-11e7-8028-09a329e7a0a1.png)

Thank you so much for your help.

**Comment by keithito** [comment]
Thanks for the additional information. As you mentioned, it looks like all of your input characters were getting stripped out of the text.

I recently updated the code to provide better options for languages other than English. There is some documentation here: https://github.com/keithito/tacotron/blob/master/TRAINING_DATA.md#non-english-data

There are 2 options for handing non-English data now:

1. You can pass the flag `--hparams=cleaners=transliteration_cleaners` to have the [Unidecode](https://pypi.python.org/pypi/Unidecode) library transliterate your text to ASCII. For example, it converts the string "1|﻿म कमल हो|म कमल हो" to "1|m kml ho|m kml ho". I don't know if this is good enough for your language, but if it is, this is the simplest option.

2. You can define your own character set. Similar to what you were doing above, you can edit symbols.py and set `_characters` to be the set of characters in your language.
https://github.com/keithito/tacotron/blob/c724ac2/text/symbols.py#L11

The new code should handle any missing symbols more gracefully instead of crashing. Note that for this option, you will also want to pass the flag `--hparams=cleaners=basic_cleaners` to train.py and eval.py.

Please let me know if this helps or if you have any other questions.

**Comment by keithito** [comment]
Closing out some inactive issues. Please feel free to re-open if there's an update.
